pkg/platform/abstract/platform_test.go

```diff
@@ -1088,6 +1089,81 @@ func (suite *AbstractPlatformTestSuite) TestEnrichAndValidateFunctionTriggers()
 			createFunctionOptions.FunctionConfig.Spec.Triggers)
 	}
 }
+func (suite *AbstractPlatformTestSuite) TestEnrichEnvVars() {
+
+	// we do our docker image test coverage on functionConfig.Spec.Build.Image but other fields, like
+	// functionConfig.Spec.Image are going through the same validation
+	for _, testCase := range []struct {
+		name            string
+		PlatformEnvFrom v1.EnvFromSource
+		FunctionEnvFrom []v1.EnvFromSource
+		ExpectedEnvFrom []v1.EnvFromSource
+	}{
+		{
+			name: "only-platform-config-is-set",
+			PlatformEnvFrom: v1.EnvFromSource{
+				ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test"}},
+			},
+			ExpectedEnvFrom: []v1.EnvFromSource{
+				{
+					ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test"}},
+				},
+			},
+		},
+		{
+			name: "only-platform-config-is-set",
+			PlatformEnvFrom: v1.EnvFromSource{
+				ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test"}},
+			},
+			FunctionEnvFrom: []v1.EnvFromSource{
+				{
+					ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test2"}},
+				},
+			},
+			ExpectedEnvFrom: []v1.EnvFromSource{
+				{
+					ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test"}},
+				},
+				{
+					ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test2"}},
+				},
+			},
+		},
+		{
+			name: "only-platform-config-is-set",
+			PlatformEnvFrom: v1.EnvFromSource{
+				ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test"}},
+			},
+			FunctionEnvFrom: []v1.EnvFromSource{
+				{
+					ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test2"}},
+				},
+			},
+			ExpectedEnvFrom: []v1.EnvFromSource{
+				{
+					ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test"}},
+				},
+				{
+					ConfigMapRef: &v1.ConfigMapEnvSource{LocalObjectReference: v1.LocalObjectReference{Name: "test2"}},
+				},
+			},
+		},
```

Comment:
how is this different than the previous test case?

Reply:
@TomerShor will remove